nabs/tests/test_plans.py

```diff
@@ -0,0 +1,207 @@
+import logging
+from collections import defaultdict
+
+import numpy as np
+import pytest
+from ophyd.device import Component as Cpt
+from ophyd.signal import Signal
+from pcdsdevices.pseudopos import DelayBase
+from pcdsdevices.sim import FastMotor
+
+import nabs.plans as nbp
+
+PLAN_TIMEOUT = 60
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.timeout(PLAN_TIMEOUT)
+def test_duration_scan(RE, hw):
+    """
+    Run the duration scan and check the messages it creates.
+    """
+    logger.debug('test_duration_scan')
+
+    # These will generate as many messages as they can in 0.01s
+    scan1 = list(nbp.duration_scan([hw.det], hw.motor, [0, 1], duration=0.01))
+    scan2 = list(nbp.duration_scan([hw.det1, hw.det2], hw.motor1, [-1, 1],
+                                   hw.motor2, [-2, 2], duration=0.01))
+
+    # I won't check behavior, but they should not error out
+    RE(scan1)
+    RE(scan2)
+
+    # Scan should cycle through positions
+    scan1_moves = list(msg.args[0] for msg in scan1 if msg.command == 'set')
```

Comment:
I like this pattern

Reply:
:+1: we have the luxury of doing things like this because we don't have to test `bluesky` here! We just have to make sure our scans have the correct messages.